src/mongocxx/uri.cpp

```diff
@@ -129,5 +130,160 @@ class write_concern uri::write_concern() const {
         stdx::make_unique<write_concern::impl>(libmongoc::write_concern_copy(wc)));
 }
 
+static stdx::optional<stdx::string_view> _string_option(mongoc_uri_t* uri, std::string opt_name) {
+    const char* value;
+
+    value = libmongoc::uri_get_option_as_utf8(uri, opt_name.c_str(), NULL);
+    if (!value) {
+        return {};
+    }
+
+    return stdx::string_view{value};
```

Comment:
I'm surprised this doesn't need to be `return {stdx::string_view{value}};`

Reply:
Huh, yeah I wasn't sure either. But I think it corresponds to constructor 8 on cppreference
https://en.cppreference.com/w/cpp/utility/optional/optional

```
template < class U = value_type >
constexpr optional( U&& value );
```